NSSServices/Controllers/VariablesController.cs

```diff
@@ -64,14 +69,136 @@ public override async Task<IActionResult> Get(int id)
             try
             {
                 if(id<0) return new BadRequestResult(); // This returns HTTP 404
-                return Ok(await agent.GetVariable(id));
+                return Ok(agent.GetVariableWithUnits(id));
             }
             catch (Exception ex)
             {
                 return await HandleExceptionAsync(ex);
             }
         }
-        
+
+        [HttpPost(Name = "Add Variable")]
+        [Authorize(Policy = Policy.AdminOnly)]
+        [APIDescription(type = DescriptionType.e_link, Description = "/Docs/Variables/Add.md")]
+        public async Task<IActionResult> Post([FromBody] VariableWithUnit entity)
+        {
+            try
+            {
+                if (!isValid(entity)) return new BadRequestResult(); // This returns HTTP 404
+
+                Variable tempV = new Variable
+                {
+                    UnitTypeID = entity.UnitTypeID,
+                    Comments = "Default unit"
+                };
+
+                VariableType tempVT = new VariableType
+                {
+                    Name = entity.Name,
+                    Code = entity.Code,
+                    Description = entity.Description
+                };
+
+                var temp = await shared.Add(tempVT);
+                tempV.VariableTypeID = temp.ID;
+                await agent.Add(tempV);
+
+                return Ok(temp);
+            }
+            catch (Exception ex)
+            {
+                return await HandleExceptionAsync(ex);
+            }
+        }
+
+        [HttpPost("[action]", Name = "Variable Batch Upload")]
+        [Authorize(Policy = Policy.AdminOnly)]
+        [APIDescription(type = DescriptionType.e_link, Description = "/Docs/Variables/Batch.md")]
+        public async Task<IActionResult> Batch([FromBody] List<VariableWithUnit> entities)
+        {
+            try
+            {
+                entities.ForEach(e => e.ID = 0);
+                if (!isValid(entities)) return new BadRequestObjectResult("Object is invalid");
+
+                List<VariableType> tempVTList = new List<VariableType>();
```

Comment:
same as above, there are a lot of items named "temp" + abbreviation here and it just doesn't read as smoothly, I have to in my mind translate to Variable/VariableType, etc.

Reply:
Variables have been renamed.